tests/ops/op_math/test_control.py

```diff
@@ -35,6 +33,8 @@ def make_ops():
         # Execute controlled version.
         cmake_ops()
 
+    expanded_tape = tape.expand()

```

Comment:
Why do we need to expand the tape?

Reply:
Otherwise, we would have `Controlled(RX(0.123, 0), control_wires=1)`.  Upon expansion, that operation is converted into `CRX`.

Maybe we could more eagerly convert a Controlled RX into `CRX`, but that's just a potential task for later.

If you notice below, the old code did request `expanded = ctrl_op.expand()`.

Reply:
Is this conversion really an "expansion"? I feel like this should be in the grouping functionality.

On the other hand, will we remove the `CRX` and all other `C...` gates?

Reply:
"expansion" is kind of a catch-all for "convert to a form more likely to be supported by a device".  

I have been thinking about doing this type of conversion more eagerly instead of waiting for the decomposition to be requested, but that might be a later PR.